build-tools/packages/build-cli/src/commands/release/report-unreleased.ts

```diff
@@ -0,0 +1,229 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import fetch from "node-fetch";
+import { promises as fsPromises } from "fs";
+import { Flags } from "@oclif/core";
+import { Logger } from "@fluidframework/build-tools";
+import { isInternalVersionRange } from "@fluid-tools/version-tools";
+import { BaseCommand } from "../../base";
+import { PackageCaretRange } from "../../lib";
+
+interface IBuildDetails {
+	definition: { name: string };
+	status: string;
+	result: string;
+	sourceBranch: string;
+	finishTime: string;
+	buildNumber: string;
+}
+
+export class UnreleasedReportCommand extends BaseCommand<typeof UnreleasedReportCommand> {
+	static description = `Creates a release report for the most recent build published to an internal ADO feed. It does this by finding the most recent build in ADO produced from a provided branch, and creates a report using that version. The report always uses the "caret" report format.`;
+
+	static flags = {
+		repoName: Flags.string({
+			description: "Repository name",
+			required: true,
+		}),
+		ado_pat: Flags.string({
+			description: "ADO Personal Access Token",
+			required: true,
+			env: "ADO_PAT",
+		}),
+		sourceBranch: Flags.string({
+			description: "Branch name across which the dev release manifest should be generated.",
+			required: true,
+		}),
+		...BaseCommand.flags,
+	};
+
+	public async run(): Promise<void> {
+		const flags = this.flags;
+
+		const PACKAGE_NAME = "@fluidframework/container-runtime";
+		const GITHUB_RELEASE_URL = `https://api.github.com/repos/microsoft/${flags.repoName}/releases`;
+		const ADO_BASE_URL = `https://dev.azure.com/${flags.repoName.toLowerCase()}/internal/_apis/build/builds?api-version=7.0`;
+		const REGISTRY_URL = `https://pkgs.dev.azure.com/${flags.repoName.toLowerCase()}/internal/_packaging/build/npm/registry/`;
+
+		// Authorization header for Azure DevOps
+		const authHeader = `Basic ${Buffer.from(`:${flags.ado_pat}`).toString("base64")}`;
+
+		if (authHeader === undefined || authHeader === null) {
+			this.error("Check your ADO Personal Access Token. It maybe incorrect or expired.");
+		}
+
+		try {
+			const buildNumber = await getFirstSuccessfulBuild(
+				authHeader,
+				ADO_BASE_URL,
+				flags.sourceBranch,
+				this.logger,
+			);
+			if (buildNumber !== undefined) {
+				this.log(
+					`Most successful build number for the last 24 hours for ${flags.sourceBranch} branch: ${buildNumber}`,
```

Comment:
What does "most successful" mean?

Reply:
The latest build across the provided branch which succeeds

Reply:
Got it. Maybe this?

```suggestion
					`Most recent successful build within the last 24 hours for ${flags.sourceBranch} branch: ${buildNumber}`,
```